python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -26,18 +26,25 @@ from cudf._lib.cpp.wrappers.timestamps cimport (
     timestamp_us,
     timestamp_ns
 )
+from cudf._lib.cpp.wrappers.durations cimport(
+    duration_s,
+    duration_ms,
+    duration_us,
+    duration_ns
+)
 from cudf._lib.cpp.scalar.scalar cimport (
     scalar,
     numeric_scalar,
     timestamp_scalar,
+    duration_scalar,
     string_scalar
 )
 cimport cudf._lib.cpp.types as libcudf_types
 
 
 cdef class Scalar:
 
-    def __init__(self, value, dtype=None):
+    def __init__(self, value, dtype=None, valid=None):
```

Comment:
I'm not sure about whether we should introduce a `valid` parameter here, as it is redundant with `value=None`. This opens up the possibility for ambiguities like `Scalar(None, valid=True)`.

`NaT` should never be considered "invalid" or "null". It's a completely valid value in its own right. If someone wants to construct a null timedelta, then there should be only one way to do that:

```
Scalar(None, dtype=np.timedelta64(...))
```

Reply:
One alternative is to make `value` an optional argument, in which case, the only way to construct a null scalar should be:

```python
Scalar(dtype=np.timedelta64(...), valid=False)
```

Then, it should be an error to pass `None` as a value.

Reply:
`NaT` is a sentinel value for Pandas to express a `null` time. PyArrow doesn't support it and neither should we since we have a true `null`.